Sources/NIOCore/AsyncSequences/NIOAsyncWriter.swift

```diff
@@ -0,0 +1,750 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Atomics
+import NIOConcurrencyHelpers
```

Comment:
Can we move these into the compiler guards?

Reply:
We can but we are not aligned on this. Even in my previous `NIOAsyncSequenceProducer` PR we haven't done this.

Reply:
IMHO if it isn't required it shouldn't be imported: if we don't have Swift > 5.5.2 then we shouldn't import these